test/fixtures/rules/std_rule_cases/L070.yml

```diff
@@ -0,0 +1,482 @@
+rule: L070
+
+test_pass_known_number_of_result_columns_1:
+  pass_str: |
+    select a, b
+    from t
+    union all
+    select c, d
+    from k
+
+test_fail_unknown_number_of_result_columns_1:
+  fail_str: |
+    select a
+    from t
+    union all
+    select c, d
+    from k
+
+test_pass_known_number_of_result_columns_2:
+  desc: Columns are specified in CTE so * in final query will return only columns specified earlier.
+  pass_str: |
+    with cte as (
+        select
+            a, b
+        from
+            t
+    )
+    select * from cte
+    union
+    select c, d
+    from t2
+
+test_fail_unknown_number_of_result_columns_2:
+  fail_str: |
+    with cte as (
+        select
+            a, b, c
+        from
+            t
+    )
+    select *
+    from cte
+    union
+    select d, e
+    from t
+
+# pass3;
+# fail3;
+test_pass_known_number_of_result_columns_3:
+  pass_str: |
+    with cte as (
+        select
+            *
+        from
+            t
+    )
+    select a, b from cte
+
+test_pass_known_number_of_result_columns_4:
+  pass_str: |
+    with cte1 as (
+        select
+            *
+        from
+            t
+    ),
+    cte2 as (
+        select
+            a, b
+        from
+            t
+    )
+    select
+        cte1.a
+        , cte1.d
+        , cte2.*
+    from cte1
+    join cte2
+    using (a)
+
+    union
+    select e, f, g, h
+    from cte3
+
+test_fail_unknown_number_of_result_columns_3:
+  fail_str: |
+    with cte1 as (
+        select
+            *
+        from
+            t
+    ),
+    cte2 as (
+        select
+            a, b
+        from
+            t
+    )
+    select
+        cte1.a,
+        cte2.*
+    from cte1
+    join cte2
+    using (a)
+    union
+    select e, f
+    from cte3
+
+test_pass_known_number_of_result_columns_5:
+  pass_str: |
+    select * from (
+        select
+            a, b
+        from
+            t
+    )
+    union
+    select c, d
+    from t2
+
+
+test_pass_known_number_of_result_columns_6:
+  pass_str: |
+    with cte2 as (
+      select b
+      from t1
+    )
+    select
+        t_alias.*
+    from cte1
+    join (select a from t) as t_alias
+    using (a)
+    union
+    select *
+    from cte2
+
+test_fail_unknown_number_of_result_columns_4:
+  fail_str: |
+    select
+        t_alias.*
+    from t1
+    join (select a from t) as t_alias
+    using (a)
+    union
+    select a,b
+    from t2
+
+
+test_pass_known_number_of_result_columns_7:
+  pass_str: |
+    select a from t1
+    union all
+    select b from t2
+
+test_pass_unknown_wildcard_number_of_result_columns_8:
+  pass_str: |
+    select a from t1
+    union all
+    select * from t2
```

Comment:
I wonder how this query behaves with [L044](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.rules.Rule_L044): "Query produces an unknown number of result columns." That rule may report an issue, but arguably we know it's okay for the same reason the query passes this rule (i.e. it must return one column or the `UNION ALL` would fail at runtime).

Nothing for you to do here, just thinking...